src/__tests__/encoder.ts

```diff
@@ -0,0 +1,17 @@
+import { Hex } from 'viem';
+
+import { Builder, IBuilder } from '@/__tests__/builder';
+
+export interface IEncoder<T, E = Hex> extends IBuilder<T> {
```

Comment:
Since this is an extension of the `IBuilder` class and it's going to be mostly used for building payloads for testing, I would rename this to `IEncoderBuilder`

Reply:
Renamed in f7bfad5d54cfc2a28765809536b4989f838b2626.